Assets/MRTK/Core/Utilities/StandardShader/ProximityLight.cs

```diff
@@ -242,34 +240,40 @@ private static void UpdateProximityLights(bool forceUpdate = false)
                 return;
             }
 
-            for (int i = 0; i < proximityLightCount; ++i)
+            int lightIndex, queueIndex;
+            for (lightIndex = queueIndex = 0; queueIndex < activeProximityLights.Count && lightIndex < proximityLightCount; ++queueIndex)
             {
-                ProximityLight light = (i >= activeProximityLights.Count) ? null : activeProximityLights[i];
-                int dataIndex = i * proximityLightDataSize;
+                ProximityLight light = activeProximityLights[queueIndex];
 
-                if (light)
-                {
-                    proximityLightData[dataIndex] = new Vector4(light.transform.position.x,
-                                                                light.transform.position.y,
-                                                                light.transform.position.z,
-                                                                1.0f);
-                    float pulseScaler = 1.0f + light.pulseTime;
-                    proximityLightData[dataIndex + 1] = new Vector4(light.Settings.NearRadius * pulseScaler,
-                                                                    1.0f / Mathf.Clamp(light.Settings.FarRadius * pulseScaler, 0.001f, 1.0f),
-                                                                    1.0f / Mathf.Clamp(light.Settings.NearDistance * pulseScaler, 0.001f, 1.0f),
-                                                                    Mathf.Clamp01(light.Settings.MinNearSizePercentage));
-                    proximityLightData[dataIndex + 2] = new Vector4(light.Settings.NearDistance * light.pulseTime,
-                                                                    Mathf.Clamp01(1.0f - light.pulseFade),
-                                                                    0.0f,
-                                                                    0.0f);
-                    proximityLightData[dataIndex + 3] = light.Settings.CenterColor;
-                    proximityLightData[dataIndex + 4] = light.Settings.MiddleColor;
-                    proximityLightData[dataIndex + 5] = light.Settings.OuterColor;
-                }
-                else
+                if (!light)
```

Comment:
Can we keep the original pattern of 

`if (light)` rather than `if (!light)` as far as keeping the code understandable.